### DISCUSSION TITLE
Set current room temperature with a home assistant-sensor

### DISCUSSION THREAD

**Discussion by ajpohv** [emsesp/EMS-ESP32#2080] [Q&A] [answered]
Instead of having my thermostat (Thermostat | RC100/Moduline 1000/1010) indicate what the room temperature is, I would like to have a different sensor from my Home Assistant instance communicate the room temperature to the boiler.

I know that it’s possible to set this up using external (Dallas) temperature sensors, but that would require rewiring everything in my house. It would be great if I could, for example, post a topic via MQTT that determines the room temperature value.

Can you help me?

**Comment by MichaelDvP** [comment, marked as answer]
> I would like to have a different sensor from my Home Assistant instance communicate the room temperature to the boiler.

Not possible. The boiler does not have any room temperature. You can set setpoints for the flowtemp on the header and dhw temperature and the boiler controls burner and pumps to achieve this temperatures.
On a thermostat you set the setpoint for roomtemperature and this device calculates the needed flowtemp from heatingcurve (measured outdoor temperature) or with PI/PID control from measured roomtemp and sends this flowtemp-setpoint to the boiler. 
So you need a thermostat if you like to send the roomtemp, or you calculate the flowtemp from your roomtemp and send it to the boiler. You can try to do the needed calculations in the ems-esp scheduler/custom entities, or in HA.

**Reply by proddy** [reply]
See also https://emsesp.org/FAQ/#can-ems-esp-simulate-a-thermostat

**Reply by ajpohv** [reply]
Thanks!

**Reply by MarkusMit** [reply]
> See also https://emsesp.org/FAQ/#can-ems-esp-simulate-a-thermostat

The correct link is: https://docs.emsesp.org/FAQ/#can-ems-esp-simulate-a-thermostat (note the 'docs.') ;-)

**Comment by proddy** [comment]
moving to a discussion
